cmd/template/httpRoutes.go

```diff
@@ -77,5 +80,39 @@ func (s *Server) handler(w http.ResponseWriter, r *http.Request) {
 
 	w.Write(jsonResp)
 }
+
+func (s *Server) pingPongWebsocketHandler(w http.ResponseWriter, r *http.Request) {
+	socket, err := websocket.Accept(w, r, nil)
+
+	if err != nil {
+		log.Print("could not open websocket")
+		// pray this works for your user
+		_, _ = w.Write([]byte("could not open websocket"))
+		w.WriteHeader(http.StatusInternalServerError)
+		return
+	}
+
+	ctx := r.Context()
+	for {
+		msgType, socketBytes, err := socket.Read(ctx)
+
+		if err != nil {
+			log.Print("could not read from websocket")
+			return
+		}
+
+		if string(socketBytes) == "PING" {
+			if err := socket.Write(ctx, msgType, []byte("PONG")); err != nil {
+				log.Print("could not write to socket")
+				return
+			}
+		} else {
+			if err := socket.Write(ctx, msgType, []byte("HUH?")); err != nil {
+				log.Print("could not write to socket")
+				return
+			}
+		}
+	}
+}
```

Comment:
i just thought of something:
since this is basically copy and pasted from one to the other, it makes more sense just to create one template file that contains the logic for all of them 
For example: https://github.com/Melkeydev/go-blueprint/blob/main/cmd/template/main.go#L7 
is used across the majority of the set up except the fiber one